go/enclave/rpcencryptionmanager/rpc_encryption_manager.go

```diff
@@ -125,17 +125,23 @@ func (rpc *RPCEncryptionManager) EncryptTxReceiptWithViewingKey(address gethcomm
 
 // DecryptTx decrypts an L2 transaction encrypted with the enclave's public key.
 func (rpc *RPCEncryptionManager) DecryptTx(encryptedTx common.EncryptedTx) (*common.L2Tx, error) {
+	if !rpc.viewingKeysEnabled {
```

Comment:
@BedrockSquirrel I think it's cleaner to move this check down into `extractTxFromBinary`, and just return the bytes from there if there's no decryption needed.